File: app/src/main/java/bonoj/me/athenaeum/root/ApplicationModule.java
package bonoj.me.athenaeum.root;

import android.content.Context;

import javax.inject.Singleton;

import bonoj.me.athenaeum.data.BooksDataSource;
import bonoj.me.athenaeum.data.source.remote.BooksRemoteRepository;
import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private final AthenaeumApplication application;

    private BooksRemoteRepository booksRemoteRepository;

    public ApplicationModule(AthenaeumApplication application){
        this.application = application;

        booksRemoteRepository = new BooksRemoteRepository(application);
    }

    @Provides
    @Singleton
    public Context provideContext(){
        return application;
    }

    @Provides
    @Singleton
    BooksDataSource provideConnectionsRepository(Context context) {
        //return new ConnectionsLocalRepository(context);
        return booksRemoteRepository;
    }
}

File: app/src/main/java/bonoj/me/athenaeum/root/ApplicationModule.kt
package bonoj.me.athenaeum.root

import android.content.Context
import bonoj.me.athenaeum.data.source.BooksDataSource
import bonoj.me.athenaeum.data.source.BooksRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: AthenaeumApplication) {

    private val booksRepository: BooksRepository

    init {
        booksRepository = BooksRepository(application)
    }

    @Provides
    @Singleton
    fun provideApplicationContext(): Context {
        return application
    }

    @Provides
    @Singleton
    fun provideBooksRemoteRepository(context: Context): BooksDataSource {
        return booksRepository
    }
}
